src/core/MIGRATION.md

```diff
@@ -687,3 +689,25 @@ There is essentially no code that _can't_ exist in a plugin. When in doubt, put
 After plugins, core is where most of the rest of the code in Kibana will exist. Functionality that's critical to the reliable execution of the Kibana process belongs in core. Services that will widely be used by nearly every non-trivial plugin in any Kibana install belong in core. Functionality that is too specialized to specific use cases should not be in core, so while something like generic saved objects is a core concern, index patterns are not.
 
 The packages directory should have the least amount of code in Kibana. Just because some piece of code is not stateful doesn't mean it should go into packages. The packages directory exists to aid us in our quest to centralize as many of our owned dependencies in this single monorepo, so it's the logical place to put things like Kibana specific forks of node modules or vendor dependencies.
+
+
+## How to
+
+### Configure plugin
+Kibana provides ConfigService if a plugin developer may want to support adjustable runtime behavior for their plugins. Access to Kibana config in New platform has been subject to significant refactoring.
+In order to have access to a config, plugin *should*:
+- Declare plugin specific "configPath" (will fallback to plugin "id" if not specified) in `kibana.json` file.
+- Export schema validation for config from plugin's main file. Schema is mandatory. If a plugin reads from the config without schema declaration, ConfigService will throw an error.
+```js
+import { schema } from '@kbn/config-schema';
+export const plugin = ...
+export const config = {
+  schema: schema.object(...),
+};
+```
+- Read config value exposed via initializerContext. No config path is required.
+```js
+const config$ = this.initializerContext.config.create();
+// or if config is optional:
+const config$ = this.initializerContext.config.createIfExists();
+```
```

Comment:
I'm assuming that `initializerContext` means the `PluginInitializerContext` that plugins receive when they are constructed. Can we make that more explicit here? I think the `this.initializerContext` makes this seem a bit magical. I'm also not sure we should encourage plugin developers to hold on to a reference to the initializerContext.

Reply:
how about the same pattern that we use in core services?
```typescript
constructor(initializerContext: PluginInitializerContext) {
  this.config$ = initializerContext.config.create();
}
```